### ISSUE TITLE
Add "input" label mode to tf.keras.preprocessing.image_dataset_from_directory

### ISSUE THREAD

**Issue by FTC55** [tensorflow/tensorflow#48219]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.4.0
- Are you willing to contribute it (Yes/No): No

**Describe the feature and the current behavior/state.**
Currently, there is no way to get `tf.keras.preprocessing.image_dataset_from_directory` to create a dataset the labels of which are the images themselves, for use in autoencoders. The behavior would be the same as that of `tf.keras.preprocessing.image.ImageDataGenerator.flow_from_directory` with the argument `class_mode="input"`. Perhaps the preexisting argument `label_mode` could be modified to include an `"input"` option.

**Will this change the current api? How?**
Minimal change to `tf.keras.preprocessing.image_dataset_from_directory` is required. Because of the nature of what is essentially a very small addition, it would be extremely unlikely to pose a compatibility issue.

**Who will benefit with this feature?**
Users wishing to benefit from the advantages of the `tf.data` API to train their image data autoencoders.

**Any Other info.**

**Comment by jvishnuvardhan** [comment]
@FTC55 Sorry for the late response. Interesting. 
I am not sure whether this feature is usable for any other applications or use cases. Do you think this will be useful for any other application other than autoencoders?

Are you still interested in contributing? If yes, please feel free to open a PR in  [keras-team/keras repo.](https://github.com/keras-team/keras/issues) repository.

Please note that Keras development moved to keras-team/keras repository to focus on only keras. Thanks!

**Comment by FTC55** [comment]
@jvishnuvardhan Hi, I'm still interested in the implementation of this feature since I frequently work with autoencoders. However, mine was more of a suggestion as I lack the skill to carry out the implementation myself. I think the feature should be implemented for consistency with the rest of the API, as I said to match `tf.keras.preprocessing.image.ImageDataGenerator.flow_from_directory` so as to satisfy all the use cases that this feature already does here. I eventually had to circumvent this limitation for the project I was working on by abandoning `tf.data.Dataset` entirely and using multiple ImageDataGenerators which to me really hurts simplicity and ease of use.

**Comment by jvishnuvardhan** [comment]
@FTC55 Would you mind opening it in keras-team/keras repo? Just need to copy and paste.

If you are busy, i can create the same feature request in that repo? Thanks

**Comment by FTC55** [comment]
@jvishnuvardhan Sure thing. Just done it. Should I close here?

**Comment by simacaonima** [comment]
邮件已收到！谢谢！
